File: mcp/sidecar/src/archive/path_masking.ts
// path_masking.ts — workspace 경로 masking

/** 사용자 홈·민감 경로를 masking한다. */
export function maskArtifactPath(inputPath: string): string {
  let masked = inputPath;
  masked = masked.replace(/[A-Za-z]:\\Users\\[^\\]+/gi, 'C:\\Users\\***');
  masked = masked.replace(/\/Users\/[^/]+/g, '/Users/***');
  masked = masked.replace(/\\Users\\[^\\]+/gi, '\\Users\\***');
  masked = masked.replace(/[A-Za-z]:\\home\\[^\\]+/gi, 'C:\\home\\***');
  return masked;
}

/** 설정 값에 secret/token 패턴이 있으면 masking한다. */
export function maskSensitiveValue(key: string, value: string): string {
  const lower = key.toLowerCase();
  if (
    lower.includes('token') ||
    lower.includes('secret') ||
    lower.includes('api_key') ||
    lower.includes('password')
  ) {
    return '***';
  }
  if (/api[_-]?key/i.test(value) || /bearer\s+/i.test(value)) {
    return '***';
  }
  return value;
}

/** 응답 JSON에 절대경로가 포함되지 않았는지 검사한다. */
export function containsAbsolutePathLeak(text: string): boolean {
  return /[A-Za-z]:\\Users\\[^\\]+/i.test(text) || /\/Users\/[^/]+/i.test(text);
}

File: app/flutter_app/lib/domain/utils/path_masking.dart
// path_masking.dart — RC artifact 경로 masking 유틸

/// 사용자 홈·민감 경로를 masking한다.
String maskArtifactPath(String path) {
  var masked = path;
  masked = masked.replaceAllMapped(
    RegExp(r'[A-Za-z]:\\Users\\[^\\]+', caseSensitive: false),
    (_) => r'C:\Users\***',
  );
  masked = masked.replaceAllMapped(
    RegExp(r'/Users/[^/]+'),
    (_) => '/Users/***',
  );
  masked = masked.replaceAllMapped(
    RegExp(r'\\Users\\[^\\]+', caseSensitive: false),
    (_) => r'\Users\***',
  );
  masked = masked.replaceAllMapped(
    RegExp(r'[A-Za-z]:\\home\\[^\\]+', caseSensitive: false),
    (_) => r'C:\home\***',
  );
  return masked;
}

/// export 본문에 secret/token/api key 패턴이 없는지 검사한다.
bool exportContainsSensitivePatterns(String text) {
  final lower = text.toLowerCase();
  const blocked = [
    'api_key',
    'api key',
    'secret_token',
    'secret token',
    'bearer ',
    'password=',
  ];
  for (final token in blocked) {
    if (lower.contains(token)) return true;
  }
  return false;
}
